sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -126,15 +126,16 @@ static az_result root(az_event_policy* me, az_event event)
  * and passes the event for the ready state to handle
  */
 AZ_INLINE az_result send_to_ready_if_topic_matches(
-    az_mqtt5_rpc_client* this_policy,
+    az_mqtt5_rpc_client_hfsm* this_policy,
     az_event event,
     az_event_policy_handler source_state)
 {
   az_mqtt5_recv_data* recv_data = (az_mqtt5_recv_data*)event.data;
+  az_mqtt5_rpc_client_codec_request_response request_response;
 
   // Ensure pub is of the right topic
-  if (_az_span_topic_matches_filter(
-          this_policy->_internal.rpc_client_codec->_internal.subscription_topic, recv_data->topic))
+  if (az_mqtt5_rpc_client_codec_parse_received_topic(
```

Comment:
Does this return a bool?

Reply:
It returns AZ_OK, added the explicit check for AZ_OK

Reply:
nit: az_result_succeeded